udemy_exercises/songs/Album.java

```diff
@@ -1,19 +1,54 @@
-package udemy_exercises.songs;
+package com.adamrice;
 
-class Album {
+import java.util.ArrayList;
+import java.util.LinkedList;
 
-}
+public class Album {
+    private String name;
+    private String artist;
+    private ArrayList<Song> songs;
+
+    public Album(String name, String artist) {
+        this.name = name;
+        this.artist = artist;
+        this.songs = new ArrayList<Song>();
+    }
+
+    public boolean addSong(String title, double duration) {
+        if (findSong(title) == null) {
+            this.songs.add(new Song(title, duration));
+            return true;
+        }
+        return false;
+    }
+
+    private Song findSong(String title) {
+        for (Song checkedSong : this.songs) {
+            if (checkedSong.getTitle().equals(title)) {
+                return checkedSong;
+            }
+        }
+        return null;
+    }
 
-/*
-**Song Player w/ Java Classes**
-     - **Album class** 
-       - Playlist of **Songs**.
-       - addPlayist() (songs must appear in the list in the order by which they were added)
-       - stop()
-       - nextSong()
-       - previousSong()
-       - repeatSong()
-       - showPlaylist()
-       - removePlaylist()
-     - **A song must exist in an album before it can be added to the playlist (so you can only play songs that you own.)**
-*/
\ No newline at end of file
+    public boolean addToPlayList(int trackNumber, LinkedList<Song> playList) {
+        int index = trackNumber - 1;
+        if ((index > 0) && (index <= this.songs.size())) {
+            playList.add(this.songs.get(index));
+            return true;
+        }
+        System.out.println("This album does not have a track " + trackNumber);
+        return false;
+    }
+
+    public boolean addToPlayList(String title, LinkedList<Song> playList) {
```

Comment:
Concerning both 'addToPlayList' methods:  though this follows the way the exercise was described, it doesn't seem appropriate for an 'Album' class to handle 'PlayList' behaviors.  In this case, the only behavior is 'add' to play list, so the other 'play list' behaviors exist somewhere else.  How might SRP be applied here to improve the Cohesion of the entire system?